pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -165,3 +176,49 @@ func (a Adapter) generateAndGetContainerConfig(componentName string, comp versio
 	containerConfig := a.Client.GenerateContainerConfig(*comp.Image, comp.Command, comp.Args, envVars, containerLabels)
 	return containerConfig
 }
+
+func (a Adapter) generateAndGetHostConfig() (container.HostConfig, error) {
+	portMap, err := getPortMap()
+	if err != nil {
+		return container.HostConfig{}, err
+	}
+	hostConfig := container.HostConfig{}
+	if len(portMap) > 0 {
+		hostConfig = a.Client.GenerateHostConfig(false, false, portMap)
+	}
+	return hostConfig, nil
+}
+
+func getPortMap() (nat.PortMap, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return nat.PortMap{}, err
+	}
+	envInfo, err := envinfo.NewEnvSpecificInfo(dir)
+	if err != nil {
+		return nat.PortMap{}, err
+	}
+	urlArr := envInfo.GetURL()
+	if len(urlArr) > 0 {
+		portmap := nat.PortMap{}
+		for _, element := range urlArr {
+			if element.ExposedPort > 0 {
+				port, err := nat.NewPort("tcp", strconv.Itoa(element.Port))
+				if err != nil {
+					return nat.PortMap{}, err
+				}
+				portmap[port] = []nat.PortBinding{
+					nat.PortBinding{
+						HostIP:   LocalhostIP,
+						HostPort: strconv.Itoa(element.ExposedPort),
+					},
+				}
+				log.Info("\nApplying URL configuration")
+				log.Successf("URL %v:%v created\n", LocalhostIP, element.ExposedPort)
```

Comment:
Same as above, `\n` shouldn't be used in logs

Reply:
removed this `\n`